### DISCUSSION TITLE
Enter an income (negative expense)

### DISCUSSION THREAD

**Discussion by peterhackenberg** [danilkinkin/buckwheat#82] [Ideas]
**Is your feature request related to a problem? Please describe.**
At this point it is not possible to record an income or a negative expense. For example, you cannot model when you get bottle deposit back or you buy gas for your car or drinks for your friends and someone contributes to the cost later.

**Describe the solution you'd like**
The possibility to enter an income or negative expense.

**Describe alternatives you've considered**
In my opinion there is no viable alternative.
At the moment you need to reduce the previous expense by the amount received later, which is quite annoying. So the real costs are not tracked correctly and the amount received has to be less than the related expense.

But besides that I love the app, great work!

**Comment by Disaster2life** [comment]
I would very much like for this, I assume this is related to the wallet function? My income isn't stable or guranteed, I really like the app so I am sticking with it, but I would really like being able to add income later on, after the budget has started, or just a wallet feature.

**Comment by danilkinkin** [comment]
I move this issue to discussion because think there needs some discussion before.

For first, I think this thing not fit to app idea. If you have received an extra budget during the current period. Better save this money. And add in next period. This app about saving.

**Reply by Gukkey** [reply]
While saving money is the reason most of us use Buckwheat I think adding an option will be helpful because there are cases where users might have sudden expenditures and have to pull out positive amount and add it in the budget to keep the flow going (think it of like adding little bit cash at the month end because your initial budget was too strict and you had some extra expenses in the middle). While this may encourage some people to be more spending, this may also prevent people (like me) deleting random small expenses from app when I add some extra cash to my budget to balance the budget per day.
